com.codeaffine.eclipse.swt/src/com/codeaffine/eclipse/swt/widget/scrollable/StyledTextAdapter.java

```diff
@@ -313,6 +313,15 @@ public void setFont( Font font ) {
 
   private void initialize() {
     styledText.setParent( this );
+    // Hide native horizontal bar to avoid rendering issues with neighbouring
+    // widgets (e.g. ruler, see https://github.com/fappel/xiliary/issues/87).
+    // External code trying to unhide it would trigger a PaintEvent, hence the
+    // listener to ensure it's always hidden.
+    styledText.addPaintListener( e -> {
```

Comment:
I'm expecting the performance impact of this to be negligible: underneath the hood, `setVisible` is a fast return if the value is already false.